src/planner/constify_now.c

```diff
@@ -163,13 +154,25 @@ constify_now_expr(PlannerInfo *root, OpExpr *op)
 		 * will be repeated with exact values during execution.
 		 * Since dst switches seem to range between -1 and 2 hours we set
 		 * the safety buffer to 4 hours.
+		 * When dealing with Intervals with month component timezone changes
+		 * can result in multiple day differences in the outcome of these
+		 * calculations due to different month lengths. When dealing with
+		 * months we add a 7 day safety buffer.
```

Comment:
Is there any rationale for the particular number 7?

Reply:
so the worst diff i can think of will be when february is involved leading up to 3 day difference due to month length. So I doubled that and rounded up to full week.